add publications to readme

mhoellmann opened the pull request.

jarkenau reviewed (dismissed):
You cloud also add the citation via an extra file, see [GitHub Docs]( https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-citation-files). Then they also show up in the right info bar of the repo.

mhoellmann commented:
good remark. I added the CITATION.cff. The CITATION.cff is not really practicable for multiple publications. I just added the main publication to the cff file and the overview remains in the README
